schedule-replicas-by-topology-labels.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Schedule Replicas by Topology Labels
+category: how-to
+aliases: ['/docs-cn/dev/how-to/deploy/geographic-redundancy/location-awareness/','/docs-cn/dev/location-awareness/']
+---
+
+# Schedule Replicas by Topology Labels
+
+To improve the high availability and disaster tolerance of TiDB clusters, it is recommended that TiKV nodes are physically distributed as much as possible. For example, TiKV nodes can be distributed on different racks or even in different data zones. According to the topology information of TiKV, the PD scheduler automatically performs scheduling at the background to isolate the replicas of Regions as much as possible, thereby maximizing the capability for disaster recovery.
+
+To make this mechanism effective, you need to report the topology information of the cluster, especially the TiKV location, to PD for proper configuration during deployment. Before you begin, see [Deploy TiDB Using TiDB Ansible (Recommended)](/online-deployment-using-ansible.md) and [Deploy TiDB Using Docker](/test-deployment-using-docker.md).
+
+## Configure `labels` based on the cluster topology
+
+### Set the `label` configuration of TiKV
+
+TiKV supports binding some attributes through command-line flags or configuration files in the form of key-value pairs. The attribute is called a `label`. TiKV reports its labels to PD so we can mark the TiKV location.
+
+Assume that the topology has three structures: zone > rack > host, use these labels to set the TiKV location.
+
+Command-line flags:
+
+{{< copyable "" >}}
+
+```
+tikv-server --labels zone=<zone>,rack=<rack>,host=<host>
+```
+
+Configuration files:
+
+{{< copyable "" >}}
+
+```toml
+[server]
+labels = "zone=<zone>,rack=<rack>,host=<host>"
+```
+
+### Set the `location-labels` configuration of PD
+
+According to the previous description, labels can be any key-value pairs used to describe TiKV attributes, but PD can not distinguish the location-marking labels and the structure among them. Therefore, you need set some configuration to understand the TiKV node topology.
+
+Set the PD configuration `location-labels` through the PD configuration files.
+
+{{< copyable "" >}}
+
+```toml
+[replication]
+location-labels = ["zone", "rack", "host"]
+```
+
+After the PD cluster is initialized, you need to use the pd-ctl tool to make online changes:
```

Comment:
To configure `location-labels` after the PD cluster is initialized, ...
Or
If PD cluster is already initialized, you need to...